analysis_functions.py

```diff
@@ -0,0 +1,160 @@
+import numpy as np
+import polars as pl
+from matplotlib import pyplot as plt
+from scipy import stats
+
+def get_general_ch_statistics(filtered_df, cols_to_keep = ['dislike_count','duration','like_count','view_count'], channel = False):
+
+    '''get general statistics for all channel on a given column'''
+
+    if channel :
```

Comment:
If you want you could do this in a more automatic way. This probably makes the function a bit more robust:
```python
columns_name = filtered_df.columns
if "channel" in columns_name:
    col_to_group_by = "channel"
elif "channel_id" in columns_name:
    col_to_group_by = "channel_id"
else:
    raise KeyError("No column named 'channel' or 'channel_id' found in the dataframe")
```